src/Aspire.Hosting/Devcontainers/DevcontainerSettingsWriter.cs

```diff
@@ -62,24 +67,46 @@ public static async Task SetPortAttributesAsync(int port, string protocol, strin
         settingsContent = settings.ToString();
         await File.WriteAllTextAsync(settingsPath, settingsContent, cancellationToken).ConfigureAwait(false);
 
-        s_writeLock.Release();
+        _writeLock.Release();
 
-        static string GetSettingsPath()
+        string GetSettingsPath()
         {
             // For some reason the machine settings path is different between Codespaces and local Devcontainers
-            // so we probe for it here. We could use options to figure out which one to look for here but that
-            // would require taking a dependency on the options system here which seems like overkill.
-            if (File.Exists(CodespaceSettingsPath))
+            // so we decide which one to use here based on the options.
+            if (codespaceOptions.Value.IsCodespace)
             {
                 return CodespaceSettingsPath;
             }
-            else if (File.Exists(LocalDevcontainerSettingsPath))
+            else if (devcontainerOptions.Value.IsDevcontainer)
             {
                 return LocalDevcontainerSettingsPath;
             }
             else
             {
-                throw new DistributedApplicationException("Could not find a devcontainer settings file.");
+                throw new DistributedApplicationException("Codespaces or Devcontainer not detected.");
+            }
+        }
+
+        async Task EnsureSettingsFileExists(string path, CancellationToken cancellationToken)
+        {
+            try
+            {
+                if (!File.Exists(path))
+                {
+                    // The extra ceremony here is to avoid accidentally overwriting the file if it was
+                    // created after we checked for its existence. If the file exists when we go to write
+                    // it then we will throw and log a warning, but otherwise continue executing.
+                    using var stream = File.Open(path, FileMode.CreateNew);
+                    using var writer = new StreamWriter(stream);
+                    await writer.WriteAsync("{}".AsMemory(), cancellationToken).ConfigureAwait(false);
+                }
+            }
+            catch (IOException ex) when (ex.Message == $"The file '{path}' already exists.")
```

Comment:
I am gonna assume you tested this 😄

Reply:
I did. The race should be pretty rare as well. I was just being cautious.